modules/home-manager/gitui.nix

```
@@ -0,0 +1,17 @@
+{ config
+, lib
+, sources
+, ...
+}:
+let
+  cfg = config.programs.gitui.catppuccin;
+  enable = cfg.enable && config.programs.gitui.enable;
+in
+{
+  options.programs.gitui.catppuccin =
+    lib.ctp.mkCatppuccinOpt "gitui";
+
+  config = lib.mkIf enable {
+    programs.gitui.theme = builtins.readFile "${sources.gitui}/theme/${cfg.flavour}.ron";
```

```suggestion
    programs.gitui.theme = "${sources.gitui}/theme/${cfg.flavour}.ron";
```
isabel was right, this can just be a path

---

Hey, just double checked this in case I didn't try this.

From my testing this seems to just write the store path to `theme.ron` instead of linking to it. I suspect this is because the two checks home-manager does to detect if it is a file (`builtins.isPath` and `lib.isStorePath`) both return `false` when compared against the path. The tricky one is `builtins.isPath`, because it seems to only return `true` if the provided argument is of type `path`, but with the interpolation it ends up being a string with context.

Would appreciate any ideas you have on how to solve this

---

ah, sorry about that. `builtins.path` should be able to help here
```suggestion
    programs.gitui.theme = builtins.path {
      name = "${cfg.flavour}.ron";
      path = "${sources.gitui}/theme/${cfg.flavour}.ron";
    };
```

---

Works perfectly! Thank you!